### PR TITLE
21874-We-have-still-users-of-old-Compiler-api-evaluateinto

### PR DESCRIPTION
**Pull request by MarcusDenker**
We have still users of old Compiler api: evaluate:in:to:
https://pharo.manuscript.com/f/cases/21874/We-have-still-users-of-old-Compiler-api-evaluate-in-to

- evaluate:in:to: was already fixed
- move old API on class side to old - deprecated category
- improve class comment OpalCompiler a little
- add test for compiling a method with class not set (nil)
	Smalltalk compiler compile: 'tst 1+2'.
  (missing test from older PR)

### CODE REVIEW

File: src/OpalCompiler-Core/OpalCompiler.class.st [hunk 1]
```diff
@@ -113,7 +124,7 @@ OpalCompiler class >> evaluate: textOrString notifying: aController logged: logF
 			
 ]
 
-{ #category : #'old - public' }
+{ #category : #'old - deprecated' }
```
Comment (jecisc) [inline src/OpalCompiler-Core/OpalCompiler.class.st:127]: If it's deprecated shouldn't it have a deprecation warning and be in the Deprecated package?

Comment (MarcusDenker) [reply src/OpalCompiler-Core/OpalCompiler.class.st:127]: it is on the TODO... the problem is that we still have senders. I will do transform deprecation after that is fixed.
